src/libs/ReportUtils.js

```diff
@@ -4087,7 +4087,7 @@ function getIOUReportActionDisplayMessage(reportAction) {
     let displayMessage;
     if (originalMessage.type === CONST.IOU.REPORT_ACTION_TYPE.PAY) {
         const {IOUReportID} = originalMessage;
-        const {amount, currency} = originalMessage.IOUDetails;
+        const {amount, currency} = originalMessage.IOUDetails || originalMessage;
```

Comment:
I don't understand this.

This _at least_ requires an explanation comment. Where are `{amount, currency}` expected to reside? _Why_ are there sometimes here, sometimes here?

Reply:
@Gonals @marcaaron I approved this as it's a deploy blocker, but I didn't really get the explanation above and I still believe something should be done here. I'm just not completely sure what.

Reply:
@saranshbalyan-1234

> Both of these values were expected to reside in the root level, but for only our original issue that we fixed, only in that case it is coming inside ioudetails key

I need help understanding this. Would you re-phrase this?

> the original pr which caused this regression

Do you have a specific PR in mind?

Reply:
> Original PR -> my last PR which caused this regression

Oh, sure 👍

Still, we need to sketch a plan for removing this `originalMessage.IOUDetails || originalMessage` expression long-term. Do I understand correctly that you suggested that this inconsistency starts on the backend side?

Reply:
Oh, I'm sorry about your situation 🙁 If your health situation makes it difficult or impossible to work on this, let us know. We might put this on hold for the time of your recovery or something like that.

Reply:
When money is sent (using send money) we store there details in `IOUDetails`, when money request is paid, the amount and currency is in the `originalMessage` field

Reply:
Huh? That seems weird 😄 Is there some way to explicitly tell whether it's one or the other? Maybe the code should have a comment about it if we are not going to improve this.